### PR TITLE
Introduce Marshaling library

### PR DESCRIPTION
**Pull request by tstruk**
Add generic Marshaling library for TPM types

### CODE REVIEW

File: include/sapi/marshal.h [hunk 1]
```diff
@@ -414,6 +414,20 @@ TPM2B_PUBLIC_KEY_RSA_Unmarshal (
     TPM2B_PUBLIC_KEY_RSA *dest);
 
 TSS2_RC
+TPM2B_PRIVATE_KEY_RSA_Marshal (
```
Comment (flihp) [inline include/sapi/marshal.h]: Should these TPM2B_PRIVATE_KEY_RSA* function prototypes be in the commit dc58ee361125b792571fa656f1182c8bfd51af67?

File: include/sapi/marshal.h [hunk 2]
```diff
@@ -834,6 +862,20 @@ TPMS_TAGGED_PCR_SELECT_Unmarshal(
     TPMS_TAGGED_PCR_SELECT *dest);
 
 TSS2_RC
+TPMS_QUOTE_INFO_Marshal (
```
Comment (flihp) [inline include/sapi/marshal.h]: Same as last comment but should be in b41f17fd471a836ce423af1a9ef1f35780fee994 instead?

File: lib/libmarshal.map [hunk 3]
```diff
@@ -56,12 +56,16 @@
         TPM2B_ECC_PARAMETER_Unmarshal;
         TPM2B_PUBLIC_KEY_RSA_Marshal;
         TPM2B_PUBLIC_KEY_RSA_Unmarshal;
+        TPM2B_PRIVATE_KEY_RSA_Marshal;
```
Comment (flihp) [inline lib/libmarshal.map]: These should move with their prototypes to the earlier commits no?

**Comment by flihp** [comment]
I picked a few small nits in the TPMU commit (https://github.com/01org/tpm2-tss/pull/492/commits/e4fa82c9bc164234be53cd5907749c85719aa817) but have approved the overall review. If you can clean those up I'd appreciate it but on the whole this code is looking good.

File: include/sapi/marshal.h (hunk 1 above)
Comment (tstruk) [inline include/sapi/marshal.h]: done

File: include/sapi/marshal.h (hunk 2 above)
Comment (tstruk) [inline include/sapi/marshal.h]: done

**Pull request merged**